### ISSUE TITLE
prediction of a multiclass

### ISSUE THREAD

**Issue by stanislavkuskov** [qubvel/segmentation_models.pytorch#249]
Hi. I try to train multiclass segmentation with an example from [repository](https://github.com/catalyst-team/catalyst/blob/master/examples/notebooks/segmentation-tutorial.ipynb). 
In the inference, the sigmoid calculation for a single class when tensor looks like [1, w, h]
`
    mask_ = torch.from_numpy(logits[0]).sigmoid()
    mask = utils.detach(mask_ > threshold).astype("float")
`
How can I calculate for a multiclass mask when tensor looks like [9, w, h]?
My example [here](https://github.com/stanislavkuskov/otus_cv_cource/blob/master/src/otus_project_dev/face_segmentation.ipynb)

**Comment by stanislavkuskov** [comment]
Solution is `torch.argmax()`

**Comment by TimbusCalin** [comment]
@stanislavkuskov  Did you manage to make a single prediction with the Catalyst model?

I see in your code that you use the runner in order to iterate over the dataloader, and then iterate on the predictions.

Do you happen to know how to make a prediction on a single image, say loaded with opencv?

**Comment by TimbusCalin** [comment]
Sergey Kolesnikov proposed three solutions:


pred1 = next(runner.predict_loader(loader=loaders["valid"]))
pred2 = runner.predict_batch(next(iter(loaders["valid"])))
pred3 = model(next(iter(loaders["valid"]))[0].view(32, -1))

**Comment by stanislavkuskov** [comment]
@TimbusCalin I make a prediction for a single image [here](https://github.com/stanislavkuskov/otus_cv_cource/blob/focalMultyClass/src/otus_project_dev/3.%20Inference.ipynb)
I'm not sure what it's correct because it's work poorly than "runner" as I think. 
I'll come back to this someday ))

**Comment by TimbusCalin** [comment]
I am assuming that you missed some of the preprocessing that is inherent to
the predict_loader(), otherwise I do not believe there should be any
problems.

În joi, 12 nov. 2020 la 00:56, Stanislav Kuskov <notifications@github.com>
a scris:
